### PR TITLE
allow complex elwise functions

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: pytato/array.py [hunk 1]
```diff
@@ -1504,7 +1504,7 @@ def _apply_elem_wise_func(x: ArrayOrScalar, func_name: str,
 
     assert isinstance(x, Array)
 
-    if x.dtype.kind != "f":
+    if x.dtype.kind != "f" and x.dtype.kind != "c":
```
Comment (kaushikcfd) [inline pytato/array.py:1507]: For `abs` the return dtype isn't correct.

**Review by kaushikcfd — APPROVED** [review]
Thanks! Could we please also add these to the tests.

**Comment by matthiasdiener** [comment]
> Thanks! Could we please also add these to the tests.

added test

**Comment by kaushikcfd** [comment]
Hah. The tests pass :) even though we have a different result dtype for abs compared to numpy. Once you fix `abs` for complex dtypes, could you also add a check that pytato's result has the same dtype as that of numpy.

**Comment by matthiasdiener** [comment]
> Hah. The tests pass :) even though we have a different result dtype for abs compared to numpy. Once you fix `abs` for complex dtypes, could you also add a check that pytato's result has the same dtype as that of numpy.

Fixed in https://github.com/inducer/pytato/pull/79/commits/62ee559242b141dc5a61e0b5d35ad4b10de12bd4

File: pytato/array.py (hunk 1 above)
Comment (matthiasdiener) [reply pytato/array.py:1507]: Fixed in https://github.com/inducer/pytato/pull/79/commits/62ee559242b141dc5a61e0b5d35ad4b10de12bd4

**Comment by kaushikcfd** [comment]
Thanks a lot! Good to land from my end.

File: pytato/array.py [hunk 2]
```diff
@@ -1517,7 +1517,16 @@ def _apply_elem_wise_func(x: ArrayOrScalar, func_name: str,
 
 
 def abs(x: Array) -> ArrayOrScalar:
-    return _apply_elem_wise_func(x, "abs")
+    if x.dtype == np.complex64:
```
Comment (kaushikcfd) [inline pytato/array.py:1520]: Could have been simplified as:
```python
if x.dtype.kind == "c":
    result_dtype = np.empty(0, dtype=x.dtype).real.dtype
else:
    result_dtype = x.dtype
```

Comment (kaushikcfd) [reply pytato/array.py:1520]: (Optional)

Comment (matthiasdiener) [reply pytato/array.py]: Good idea, fixed.

**Comment by matthiasdiener** [comment]
Is this good to go @inducer ?

**Comment by inducer** [comment]
Thanks!

**Pull request merged**